text/0000-naming-conventions.md

```diff
@@ -0,0 +1,267 @@
+
+The inspiration comes from:
+
+https://www.alainschlesser.com/interface-naming-conventions/
+https://www.php-fig.org/psr/psr-2/
+
+# Detailed design
+
+#### Keywords and constants
+
+PHP [keywords](http://php.net/manual/en/reserved.keywords.php) must be in lower case, as well as PHP constants `true`, `false` and `null`. User defined constants must be in all upper-case with underscores separating words
+
+```php
+<?php
+define( 'ENV', 'staging' );
+```
+
+#### Namespace and use declarations
+
+Namespace must be defined at the top of the file. There must be one blank line after the `namespace` declaration. Namespaces must follow the class names naming - capitalized words separated by underscores.
+
+When present, all `use` declarations must go after the `namespace` declaration. There must be one use keyword per declaration, and there must be one blank line after the `use` block.
+
+```php
+<?php
+namespace Theme_Name\Subnamespace;
+
+use Foo_Class;
+use Bar_Class as Bar;
+use Plugin_Name\Admin as Admin;
+
+// ... additional PHP code ...
+```
+
+#### Variables and function names
+
+When naming variables, actions/filters and function names use `lowercase` letters separated by underscores.
+Don't abbreviate or obfuscate the names, code should be unambiguous and self-documenting.
+
+```php
+<?php
+$post_list = get_posts();
+
+function get_authors_names_from_posts( $posts_array ) {
+  foreach ( $posts_array as $post_object => $post ) {
+      $user_info      = get_user_by( 'ID', $post->post_author );
+      $author_names[] = $user_info->data->user_nicename;
+  }
+
+  return $author_names;
+}
+
+$authors_list = get_authors_names_from_posts( $post_list );
+```
+
+#### Classes
+
+The term 'class' refers to all classes, interfaces and traits.
+
+Class names must be capitalized words separated by underscores. Acronyms should be all upper case.
+
+```php
+<?php
+namespace Some_Name;
+
+class Some_Class {
+  // constants, properties, methods.
+}
+```
+
+```php
+<?php
+namespace Some_Name;
+
+class Particular_Class implements Thing {
+  // constants, properties, methods.
+}
+```
+
+When naming interfaces use the most meaningful name for the interface. Class name should be more qualified that provide additional information about the specific implementation of the interface it implements. For instance
+
+```php
+<?php
+namespace Templates;
+
+interface Template {
+  // methods which class must implement.
+}
+```
+
+Now we implement it in different classes
+
+```php
+<?php
+namespace Templates;
+
+class Post_Template implements Template {
+  // Specific implementation for the posts.
+}
+```
+
+```php
+<?php
+namespace Templates;
+
+class Page_Template implements Template {
+  // Specific implementation for the pages.
+}
+```
+
+The `extends` and `implements` keywords must be declared on the same line as the class name. The opening brace for the class must go in the same line as the class name.
```

Comment:
Again- there should be no code formatting guidelines in a naming conventions RFC.